molo/core/content_import/helpers/importing.py

```diff
@@ -5,178 +5,225 @@
+    language, _ = SiteLanguage.objects.get_or_create(
+        locale=Locale.parse(locale).language,
+        is_main_language=is_main)
 
-        page.subtitle = p.subtitle
-        page.body = json.dumps([
-            {'type': 'paragraph', 'value': p.description},
-            {'type': 'paragraph', 'value': p.content}
-        ])
-        is_featured = p.featured if p.featured else False
-        is_featured_in_category = p.featured_in_category \
-            if p.featured_in_category else False
-
-        page.featured_in_latest = is_featured
-        page.featured_in_homepage = is_featured_in_category
-        for tag in p.author_tags:
-            page.metadata_tags.add(tag)
-        if p.image_host and p.image:
-            page.image = get_image_file(p.image_host, p.image)
-
-        page.save_revision().publish()
-
-        return page
-
-    def update_pages_with_linked_page_field(self):
-        for p in self.ws.S(Page).all()[:10000]:
-            if p.linked_pages:
-                for lp in p.linked_pages:
-                    try:
-                        page = ArticlePage.objects.get(uuid=p.uuid)
-                        page.body.stream_data.append(
-                            {u'type': u'page',
-                             u'value': ArticlePage.objects.get(uuid=lp).pk})
-                        page.save_revision().publish()
-                    except ArticlePage.DoesNotExist:
-                        print 'Linked page does not exist %s' % lp
-
-    def import_categories_for_child_language(
-            self, category, selected_locale, site_language):
-        if category.source:
-            try:
-                parent = SectionPage.objects.get(
-                    uuid=category.source)
-                translated_section = self.import_section_content(
-                    category, site_language)
-                PageTranslation.objects.get_or_create(
-                    page=parent,
-                    translated_page=translated_section)
-            except SectionPage.DoesNotExist:
-                print "couldn't find", category.source, (
-                    SectionPage.objects.all().values('uuid'))
-        else:
-            print "no source found for: ", category.source, (
+    return {
+        'locale': locale,
+        'language': language,
+    }
+
+
+def import_locale_content(repo, lang, parent):
+    import_all_categories(repo, lang, parent)
+    import_all_pages(repo, lang)
+
+
+def main_section_datum(repo, lang):
+    # NOTE For sections that correspond to a repository (for the multi-repo
+    # import case), We can't simply rely on the ORM to create a uuid for us,
+    # since we need to retrieve the section if it already exists instead of
+    # creating it again. This means we need to find the section by its
+    # corresponding repository's name and the locale that the section was
+    # created for.  At the moment, we use the uuid field for this, though isn't
```

Comment:
I was splitting hairs here -- meant that using `uuid` is a bit wrong, since uuids are *universally* unique. True though, that using the index name as the id is unique for this instance, so sounds safe enough.